src/csv2docx/csv2docx.py

```diff
@@ -34,36 +36,75 @@ def create_unique_name(filename: str, path: Path) -> Path:
     return filepath
 
 
+def generate_docx(data: dict, template: str, mergefields: set) -> MailMerge:
+    """Generates a single docx
+    Args:
+        data: a dictionary representing a single .csv row
+        template: the name of the template .docx
+        mergefields: a set of fields to be filled by the data
+    Returns:
+        A MailMerge (.docx) object with filled values
+    """
+
+    # Must create a new MailMerge for each file
+    docx = MailMerge(template)
+    fields = {key: data[key] for key in mergefields}
+    docx.merge_templates([fields], separator="page_break")
+    return docx
+
+
 def convert(
     data: str, template: str, name: str, path: str = "output", delimiter: str = ";"
-) -> None:
-    print("Getting .docx template and .csv data files ...")
-
-    with open(data, "rt") as csvfile:
-        csvdict = csv.DictReader(csvfile, delimiter=delimiter)
-        csv_headers = csvdict.fieldnames
-        if csv_headers and name not in csv_headers:
-            print("Column name not found. Please enter valid column name")
-            exit(1)
-        docx = MailMerge(template)
-        docx_mergefields = docx.get_merge_fields()
-
-        print(f"DOCX fields : {docx_mergefields}")
-        print(f"CSV field   : {csv_headers}")
-
-        # see if all fields are accounted for in the .csv header
-        column_in_data = set(docx_mergefields) - set(csv_headers)
-        if len(column_in_data) > 0:
-            print(f"{column_in_data} is in the word document, but not csv.")
-            return
-
-        print("All fields are present in your csv. Generating Word docs ...")
-        output_path = create_output_folder(path)
-
-        for row in csvdict:
-            # Must create a new MailMerge for each file
-            docx = MailMerge(template)
-            single_document = {key: row[key] for key in docx_mergefields}
-            docx.merge_templates([single_document], separator="page_break")
-            filename = create_unique_name(row[name], output_path)
-            docx.write(filename)
+) -> bool:
+
+    try:
+        csvfile = open(data, "rt")
+    except FileNotFoundError:
+        raise FileNotFoundError(f"The data .csv file '{data}' could not be found")
+    else:
```

Comment:
I forgot to say that this was quite brilliant. I assume that we could check if the file exists (e.g., at location `/path`), but have opted for this try/except/else because then we can test the validity of the file?

Alternatively, we could have checked if the file exists via: `os.exists(data) == True`